From f1e47c25444fa6650c71d631a08c2fa96ad34416 Mon Sep 17 00:00:00 2001
From: Amir Ayub <dev.amirayub@gmail.com>
Date: Sat, 10 Oct 2026 21:51:49 +0300
Subject: [PATCH 3/3] Make the static Metal target depend on the common kernel
 sources

The static target compiled the common kernels without depending on
CommonKernels or marking CommonKernelSources.cpp generated, the two lines
sharedTarget has. With the shared plugin off, a parallel make compiled
CommonCalcConstantPotentialForce.cpp before CommonKernelSources.h existed
('CommonKernelSources.h' file not found).
---
 platforms/metal/staticTarget/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/platforms/metal/staticTarget/CMakeLists.txt b/platforms/metal/staticTarget/CMakeLists.txt
index 9ecdb87f8..2f6291c3b 100644
--- a/platforms/metal/staticTarget/CMakeLists.txt
+++ b/platforms/metal/staticTarget/CMakeLists.txt
@@ -2,9 +2,9 @@
 # Include METAL related files.
 #
 
-SET_SOURCE_FILES_PROPERTIES(${KERNELS_CPP} ${KERNELS_H} PROPERTIES GENERATED TRUE)
+SET_SOURCE_FILES_PROPERTIES(${KERNELS_CPP} ${KERNELS_H} ${COMMON_KERNELS_CPP} PROPERTIES GENERATED TRUE)
 ADD_LIBRARY(${STATIC_TARGET} STATIC ${SOURCE_FILES} ${SOURCE_INCLUDE_FILES} ${API_ABS_INCLUDE_FILES})
-ADD_DEPENDENCIES(${STATIC_TARGET} MetalKernels)
+ADD_DEPENDENCIES(${STATIC_TARGET} CommonKernels MetalKernels)
 
 TARGET_LINK_LIBRARIES(${STATIC_TARGET} ${OPENMM_LIBRARY_NAME} "-framework Foundation -framework Metal -framework QuartzCore -framework IOKit")
 SET_TARGET_PROPERTIES(${STATIC_TARGET} PROPERTIES COMPILE_FLAGS "${EXTRA_COMPILE_FLAGS} -DOPENMM_COMMON_BUILDING_STATIC_LIBRARY")
-- 
2.54.0 (Apple Git-157)

